lib/internal/encoding/single-byte.js

```diff
@@ -0,0 +1,157 @@
+  'windows-1251': [899, 1, 7191, -7111, 7115, 8, -6, 1, 139, -124, -7207, 7216, -7215, 2, -1, 4, 67, 7110, 1, 3, 1, 5, -15, 1, -8060, 8330, -7369, 7137, -7136, 2, -1, 4, -959, 878, 80, -86, -868, 1004, -1002, 1, 858, -856, 859, -857, 1, 1, 1, 857, -855, 1, 853, 80, 59, -988, 1, 1, 922, 7365, -7362, -921, 925, -83, 80, 2, -71, ...e(63)],
+  'windows-1252': [...p1, -7515, 7530, -7888, 7897, -7911, -197, 240, -238, 1, ...w1, 225, -6],
+  'windows-1253': [...p1, -8089, 8104, -8102, 8111, -8109, 1, 1, 1, 1, ...w3, 1, 1, 1, 1, 741, 1, -739, 1, 1, 1, 1, 1, 1, f, 2, 1, 1, 1, 8039, -8037, 1, 1, 1, 721, -719, 1, 1, ...i7],
+  'windows-1254': [...p1, -7515, 7530, -7888, 7897, -7911, -197, 1, 1, 1, ...w1, 1, 218, -216, ...e(47), 79, -77, ...e(11), 84, 46, -127, ...e(16), 48, -46, ...e(11), 53, 46],
+  'windows-1255': [...p1, -7515, 7530, -8102, 8111, -8109, 1, 1, 1, 1, ...w8, -7480, 7750, -8328, 8096, -8094, ...e(7), 8199, -8197, 1, 1, 1, 1, 46, -44, ...e(14), 62, -60, 1, 1, 1, 1, 1265, ...e(19), 45, 1, 1, 1, 1, ...h(7), -36, ...i8],
+  'windows-1256': [8237, -6702, 6556, -7816, 7820, 8, -6, 1, -7515, 7530, -6583, 6592, -7911, 1332, 18, -16, 39, 6505, 1, 3, 1, 5, -15, 1, -6507, 6777, -6801, 6569, -7911, 7865, 1, -6483, -1562, 1388, -1386, ...e(7), 1557, -1555, ...e(14), 1378, -1376, 1, 1, 1, 1377, 162, -160, ...e(21), -1375, 1376, 1, 1, 1, 6, 1, 1, 1, -1379, 1380, -1378, 1379, 1, 1, 1, -1377, 1, 1, 1, 1, 1374, 1, -1372, 1, 1372, 1, 1, 1, -1370, 1371, 1, -1369, 1370, -1368, 1369, -1367, 1, 7954, 1, -6461],
+  'windows-1257': [...w0, -8102, 8111, -8109, 28, 543, -527, -40, ...w3, 19, 556, -572, 1, f, 2, 1, 1, f, 2, 1, 49, -47, 173, -171, 1, 1, 1, 24, -22, ...e(5), ...p3, 347],
+  'windows-1258': [...p1, -7515, 7530, -8102, 8111, -7911, -197, 1, 1, 1, ...w8, -7480, 7750, -8328, 8096, -7911, -182, 1, 218, -216, ...e(34), 64, -62, ...e(7), 565, -563, 1, 1, 65, -63, 568, -566, 1, 204, -202, 1, 1, 1, 1, 1, 1, 211, 340, -548, 1, 1, 1, 33, -31, ...e(7), 534, -532, 1, 1, 34, -32, 562, -560, 1, 173, -171, 1, 1, 1, 1, 1, 1, 180, 7931],
+  'windows-874': [8237, -8235, 1, 1, 1, 8098, -8096, ...e(10), ...w8, -8060, ...e(8), 3425, ...e(57), f, f, f, f, 5, ...e(28), f, f, f, f],
+  'x-mac-cyrillic': [913, ...e(31), 7153, -8048, 992, -1005, 4, 8059, -8044, 848, -856, -5, 8313, -7456, 80, 7694, -7773, 80, 7627, -8557, 8627, 1, -7695, -929, 988, -137, -4, 80, -77, 80, -78, 80, -79, 80, -2, -83, -857, ...m0, 875, 80, -79, 80, -7, 7102, 1, 8, 1, -5, 1, -7970, 7975, -7184, 80, -79, 80, 7351, -7445, 80, -2, -31, ...e(30), 7262],
+};
+
+/* eslint-enable @stylistic/js/max-len */
+
+/* fallback/single-byte.js + single-byte.node.js, simplified */
+
+function getEncoding(encoding) {
+  if (encoding === 'x-user-defined') {
+    return TypedArrayFrom(Uint16Array, { length: 256 }, (_, i) => (i >= 0x80 ? 0xf700 + i : i));
```

Comment:
A doc comment link back to the spec here would be helpful. Exceedingly few people are actually familiar with this.